src/NServiceBus.Persistence.DynamoDB/SynchronizedStorage/StorageSession.cs

```diff
@@ -31,12 +31,13 @@ public void AddRange(IEnumerable<TransactWriteItem> writeItems)
 
         void CheckCapacity()
         {
-            // TODO: 25 should be a constant probably in the settings
-            if (batch.Count > 25)
-            {
-                throw new Exception(
-                    "Transactional writes are limited to 25 items. Each saga counts as one item. Outbox, if enabled, counts as one item plus one additional item for each outgoing message.");
-            }
+            // TODO: Should we really capture this on the client side? Allows for better exception messages though.
```

Comment:
Good point. Let's switch over to the service exception. It is more expressive and in case they ever relax some constraints we don't have to change this code

Reply:
to be fair the service exception is not very expressive as it's a ginormous blob of text, nowhere stating that this is about transaction items being exceeded but I still feel that there is a significant risk in trying to wrap these exceptions, especially since there doesn't seem to be a reliable way to identify the exception "type"

Reply:
discussed with Daniel: The exception is really confusing and noisy, so let's fix this check and keep it but consider throwing a Dynamo Exception. We should then also add a test to verify the limits so we're immediately noticing when the limits are changed again.